blocks/bricks/bn.py

```diff
@@ -237,23 +248,27 @@ def _allocate(self):
         # aren't in self.parameters.
         add_annotation(self.population_mean, self)
 
-        if not self.mean_only:
+        if self.learn_scale and not self.mean_only:
             # "gamma", from the Ioffe & Szegedy manuscript.
             self.scale = shared_floatx_nans(var_dim, name='batch_norm_scale',
                                             broadcastable=broadcastable)
 
             add_role(self.scale, BATCH_NORM_SCALE_PARAMETER)
             self.parameters.append(self.scale)
+        else:
+            self.scale = tensor.constant(1., dtype=theano.config.floatX)
 
+        if not self.mean_only:
             self.population_stdev = shared_floatx(numpy.ones(var_dim),
                                                   name='population_stdev',
                                                   broadcastable=broadcastable)
             add_role(self.population_stdev, BATCH_NORM_POPULATION_STDEV)
             add_annotation(self.population_stdev, self)
 
     def _initialize(self):
-        self.shift_init.initialize(self.shift, self.rng)
-        if not self.mean_only:
+        if is_shared_variable(self.shift):
```

Comment:
Wanted to avoid the situation where that flag is changed after allocation
but before initialization. Which is exotic, but better to program
defensively, no? This way it simply has no effect on the brick's operation
post-allocation, rather than doing some combination of inconsistent things.

>